tests/rewrite_test.py

```diff
@@ -67,7 +70,49 @@ def test_change_batch_size_variable_size(self):
                                        np.array([42.]).reshape([1, 1])))
         result = sess.run(result_tensor.name,
                           {input_tensor.name:
-                             np.array([0, 1]).reshape([2, 1])})
+                           np.array([0, 1]).reshape([2, 1])})
         self.assertTrue(np.array_equal(result,
                                        np.array([42., 43.]).reshape([2, 1])))
 
+  def test_change_batch_size_saved_model(self):
+    """
+    Verifies that changes of batch size survive serializing the graph as a
+    SavedModel
+    """
+    temp_dir = tempfile.mkdtemp()
+    print("Temp dir is {}".format(temp_dir))
+    tf_g = tf.Graph()
+    with tf_g.as_default():
+      input_tensor = tf.placeholder(dtype=tf.float32, shape=[32, 1],
+                                    name="Input")
+      result_tensor = input_tensor + 42.0
+      with tf.Session() as sess:
+        tf.saved_model.simple_save(sess, temp_dir + "/model_before",
+                                   inputs={"in": input_tensor},
+                                   outputs={"out": result_tensor})
+
+    # Make sure the original SavedModel loads properly
+    with tf.Session(graph=tf.Graph()) as sess:
+      tf.saved_model.load(sess, [tf.saved_model.tag_constants.SERVING],
+                          temp_dir + "/model_before")
+
+    g = gde.saved_model_to_graph(temp_dir + "/model_before")
+    gde.rewrite.change_batch_size(g, None, [g[input_tensor.name]])
+    g.to_saved_model(temp_dir + "/model_after")
+
+    with tf.Session(graph=tf.Graph()) as sess:
+      tf.saved_model.load(sess, [tf.saved_model.tag_constants.SERVING],
+                          temp_dir + "/model_after")
+      result = sess.run(result_tensor.name,
+                        {input_tensor.name:
+                         np.array([0]).reshape([1, 1])})
+      self.assertTrue(np.array_equal(result,
+                                     np.array([42.]).reshape([1, 1])))
+      result = sess.run(result_tensor.name,
+                        {input_tensor.name:
+                         np.array([0, 1]).reshape([2, 1])})
+      self.assertTrue(np.array_equal(result,
+                                     np.array([42., 43.]).reshape([2, 1])))
+
+    # Remove temp dir if the test is successful
+    shutil.rmtree(temp_dir)
```

Comment:
did you want to keep the director if the test was not successful? If not then it might be good to put this in a `finally` block.